### PR TITLE
chore: increase min supported java version to 17

### PR DESCRIPTION
**Pull request by Vovchyk**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

`fed:vovchyk/min-java-ver-bump`

### CODE REVIEW

**Comment by Vovchyk** [comment]
pipeline:run

File: rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java [hunk 1]
```diff
@@ -102,12 +99,7 @@ void blockAddressCheck_logInvalidInetAddressException() throws UnknownHostExcept
         channelManager.add(peer);
         channelManager.tryProcessNewPeers();
 
-        Logger logger = mock(Logger.class);
-        TestUtils.setFinalStatic(channelManager, "logger", logger);
-
         Assertions.assertTrue(channelManager.isAddressBlockAvailable(otherPeer.getInetSocketAddress().getAddress()));
-
-        verify(logger).error(anyString(), any(Throwable.class));
```
Comment (asoto-iov) [inline rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java:110]: so now, are not we checking if a log was written?

**Review by asoto-iov — APPROVED** [review]

**Review by nagarev — APPROVED** [review]
LGTM!

File: rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java (hunk 1 above)
Comment (Vovchyk) [reply rskj-core/src/test/java/org/ethereum/net/server/ChannelManagerImplTest.java:110]: yes. First, there's a doubt on whether we need to cover logging with unit tests. Second, in this particular case verifying if the logger receives any string seems not to be that useful. And last but not least, starting from java17 it's prohibited to set private static field via reflection (in our case it's a logger)

**Pull request merged**